src/app/loadout-builder/mod-assignments.ts

```diff
@@ -0,0 +1,140 @@
+import { DestinyEnergyType } from 'bungie-api-ts/destiny2';
+import _ from 'lodash';
+import { DimItem, PluggableInventoryItemDefinition } from '../inventory/item-types';
+import { ProcessItem, ProcessMod } from './process-worker/types';
+
+const DEFAULT_ENERGY_INVESTMENT = 1000;
+
+interface ItemAssignments {
+  mods: ProcessMod[];
+  energySwapped: boolean;
+}
+
+export class ModAssignments {
+  // We just keep track of the id's and hashes as we assume only valid items and mod
+  // combninations are passed in
```

Comment:
*combinations

If  this is documentation for `slotDependantAssignments`, use a JSDoc comment block (`/** */`) so it'll show up in autocomplete.

Reply:
These fields are private so they won't be visible outside this class.